ocaml_version.ml

```diff
@@ -385,23 +391,32 @@ let trunk_variants (arch:arch) =
   List.map (Configure_options.to_t Sources.trunk) (base @ arch_opts)
 
 let compiler_variants arch ({major; minor; _} as t) =
-  let f = List.map (Configure_options.to_t t) in
-  match major,minor,arch with
-    | 4,12,arch  -> trunk_variants arch
-    | 4,11,`X86_64 -> f [[]; [`Afl]; [`Flambda]; [`Frame_pointer]]
-    | 4,10,`X86_64 -> f [[]; [`Afl]; [`Flambda]; [`Frame_pointer]]
-    | 4,9,`X86_64 -> f [[]; [`Afl]; [`Flambda]; [`Frame_pointer]]
-    | 4,8,`X86_64 -> f [[]; [`Afl]; [`Flambda]; [`Frame_pointer]]
-    | 4,11,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,10,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,9,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,8,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,7,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,6,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,5,_ -> f [[]; [`Afl]; [`Flambda]]
-    | 4,4,_ -> f [[]; [`Flambda]]
-    | 4,3,_ -> f [[]; [`Flambda]]
-    | _ -> f [[]]
+  if major = Releases.trunk.major && minor = Releases.trunk.minor then
+    trunk_variants arch
+  else
+    let variants =
+      (* No variants for OCaml < 4.00 *)
+      if major < 4 then []
+      else
+        (* +fp for OCaml 4.08+ on x86_64 *)
+        let variants =
+          if arch = `X86_64 && (major <> 4 || minor >= 8) then
```

Comment:
are we certain those are going to be around for OCaml 5.x?

Reply:
Doesn’t matter - we’re certain they’ve been around since 4.08! 🙂 If we try to build something which no longer exists, we get a tangible error... what happened with the old approach was a few Docker images subtly missing tags.